lib/api/src/js/instance.rs

```diff
@@ -110,10 +107,17 @@ impl Instance {
             })
             .collect::<Result<Exports, InstantiationError>>()?;
 
+        // If the memory is imported then also export it for backwards compatibility reasons
+        // (many will assume the memory is always exported) - later we can remove this
+        if exports.get_memory("memory").is_err() {
+            if let Some(memory) = externs.iter().filter(|a| a.ty(store).memory().is_some()).next() {
+                exports.insert("memory", memory.clone());
+            }
+        }
```

Comment:
I really don't think we should do this. This should live somewhere else (not on the JS API layer, as it's not JS or VM specific)

Reply:
that will trigger some increased coding because of these...

![image](https://user-images.githubusercontent.com/17711537/185571191-54e44bb1-61ef-456e-8a74-0b9bf14a27df.png)

Those if we are ok to modify the interfaces that assume memory is an export (perhaps we need to put in some form of abstraction) then we can get rid of these.